### PR TITLE
Add top-level dependency to bundler conflict explanation

### PR DESCRIPTION
**Pull request by jurre**
Includes the top-level dependency when explaining a version conflict for
Bundler, to give users some more context about which dependency needs
upgrading.

### CODE REVIEW

File: bundler/spec/dependabot/bundler/update_checker/conflicting_dependency_resolver_spec.rb [hunk 1]
```diff
@@ -75,40 +75,5 @@
         expect(conflicting_dependencies).to match_array([])
       end
     end
-
-    context "with multiple blocking dependencies" do
```
Comment (jurre) [inline bundler/spec/dependabot/bundler/update_checker/conflicting_dependency_resolver_spec.rb:79]: Moved this to the native_helper spec for easier debugging

File: bundler/spec/native_helpers/functions/conflicting_dependency_resolver_spec.rb [hunk 2]
```diff
@@ -36,6 +36,95 @@
...
+      let(:gemfile_fixture_name) { "transitive_blocking" }
+      let(:lockfile_fixture_name) { "transitive_blocking.lock" }
+      let(:dependency_name) { "activesupport" }
+      let(:target_version) { "6.0.0" }
+
+      it "returns a list of dependencies that block the update" do
+        expect(conflicting_dependencies).to match_array(
+          [
+            {
+              "explanation" => "rails (5.2.0) requires activesupport (= 5.2.0) via rails (5.2.0)",
```
Comment (feelepxyz) [inline bundler/spec/native_helpers/functions/conflicting_dependency_resolver_spec.rb:49]: Should this include `via rails`?

Comment (feelepxyz) [reply bundler/spec/native_helpers/functions/conflicting_dependency_resolver_spec.rb:49]: Looks like there's some code that guards against this
```suggestion
              "explanation" => "rails (5.2.0) requires activesupport (= 5.2.0)",
```

**Review by feelepxyz — COMMENTED** [review]
LGTM 👍  added a spec suggestion that looks covered by the code `if spec.name == top_level.name`

File: bundler/helpers/lib/functions/conflicting_dependency_resolver.rb [hunk 3]
```diff
@@ -34,14 +37,37 @@ def conflicting_dependencies
...
+      parsed_lockfile.specs.filter do |spec|
+        spec.dependencies.any? do |dep|
+          dep.name == dependency_name &&
+            !dep.requirement.satisfied_by?(version)
+        end
+      end
+    end
+
+    def top_level_specs_for(parent_spec)
+      return [parent_spec] if parsed_lockfile.dependencies.key?(parent_spec.name)
```
Comment (feelepxyz) [inline bundler/helpers/lib/functions/conflicting_dependency_resolver.rb:49]: nit: extract a `top_level?(dependency_name)` method?

File: bundler/spec/native_helpers/functions/conflicting_dependency_resolver_spec.rb (hunk 2 above)
Comment (jurre) [reply bundler/spec/native_helpers/functions/conflicting_dependency_resolver_spec.rb:49]: Oh right, let me double check if it's just the spec that's wrong or some of the logic is wonkey

**Review by feelepxyz — APPROVED** [review]
🙌

**Pull request merged**